ledger/complete/mtrie/flattener/encoding_v6.go

```diff
@@ -0,0 +1,113 @@
+package flattener
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/hash"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/node"
+)
+
+// ReadNodeV6 reconstructs a node from data read from reader.
+// Scratch buffer is used to avoid allocs. It should be used directly instead
+// of using append.  This function uses len(scratch) and ignores cap(scratch),
+// so any extra capacity will not be utilized.
+// If len(scratch) < 1024, then a new buffer will be allocated and used.
+func ReadNodeV6(reader io.Reader, scratch []byte, getNode func(nodeIndex uint64) (*node.Node, error)) (*node.Node, ledger.Path, *ledger.Payload, error) {
```

Comment:
This will be used by importer to read checkpoint file and import the payloads to storage